Initial version of CSE-712 feature implementation

FedorUporov opened the pull request.

YaroslavYaroslavtsev committed:
Merge pull request #22 from nobitlost/develop

Added path to error message

Commit:
Merge pull request #50 from nobitlost/master

Added path info to error messages

Commit:
Version updated.

Commit:
Initial version of CSE-712 feature implementation

ppetrosh commented on README.md:
```diff
@@ -607,6 +607,7 @@ Lines starting with `@` followed by space or a line break are treated as comment
   * <code>--clear-cache</code> &mdash; remove cache before builder starts running.
   * <code>--cache-exclude-list <i>&lt;path_to_file&gt;</i></code> &mdash; path to exclude list file.
   * <code>--lib(s) <i>&lt;path_to_file|path_to_directory|glob&gt;</i></code> &mdash; path to JavaScript file to include as libraries
+  * <code>--suppress-duplicate-includes-warning</code> &mdash; do not warn if sources with same content were included multiple times
```
do not show a warning if a source file with the exact content was included in the multiple times from different places, that results in code duplication.

ppetrosh commented on src/cli.js:
```diff
@@ -70,6 +70,7 @@ where:
 \t\u001b[34m--clear-cache\u001b[39m - delete cache folder before running
 \t\u001b[34m--cache-exclude-list <path_to_file>\u001b[39m - path to exclude list file
 \t\u001b[34m--lib(s) <path_to_file|path_to_directory|glob>\u001b[39m - path to Javascript file to include as libraries
+\t\u001b[34m--suppress-duplicate-includes-warning\u001b[39m - hide warnings about included files with same content
```
suppress warnings about multiple includes of a file with the same exact content

ppetrosh commented on src/Machine.js:
```diff
@@ -297,6 +299,15 @@ class Machine {
     // read
     const res = this.fileCache.read(reader, includePath);
 
+    // Check if source with same hash value has already been included 
```
Is there a way to show the place in the code of the previous and the current inclusions? The warning should be actionable.

Commit:
CSE-712: Update descriptions and fix typos.

Commit:
Update @octokit API usage because of error:
[200~  Message:
    Error: [before-after-hook]: "Hook()" repurposing warning, use "Hook.Collection()". Read more: https://git.io/upgrade-before-after-hook-to-1.4
    DEPRECATED (@octokit/rest): `repos.getContent()` is deprecated, use `repos.getContents()`
    DEPRECATED (@octokit/rest): callbacks will be removed in v16. Use async/await or Promises instead.
    GitHub API rate limit exceeded (input.nut:3)

Commit:
CSE-712: Update main --suppress-duplicate-includes-warning option logic. Show both original and duplicate file.

Commit:
Fix unit test accoring new warning message format.

Commit:
Suppress duplicate warning everywhere in the unit tests to avoid stderr flood.

Commit:
Merge branch 'develop' of https://github.com/electricimp/Builder into CSE-712

Commit:
Increase the Builder version to 2.6.0.

Commit:
CSE-712: Report the place, where duplicate file was included, instead of path to duplicate file.

Commit:
CSE-712: Allow to select warning message format.

Commit:
Choice multi line warning message, code cleanup and unit-test fixes.
